### PR TITLE
Modelling d

### PR DESCRIPTION
**Pull request by zunicd**
Completed modelling notebooks and reorganized embeddings notebooks. Created folders results and saved models for modelling results. Copied best models for each task to best_models in saved_models.

### CODE REVIEW

File: notebooks/embeddings/ACP_prose_embeddings.ipynb [hunk 1]
```diff
@@ -210,17 +210,17 @@
      "output_type": "stream",
      "text": [
       "# loading the pre-trained ProSE DLM model\n",
-      "# writing: ../data/acp/prose/train/acp_train_dlm_avg.h5\n",
+      "# writing: ../../data/acp/prose/train/acp_train_dlm_avg.h5\n",
       "# embedding with pool=avg\n",
-      "# 1361 sequences processed...\r"
+      "# 1372 sequences processed...\r"
```
Comment (MichalRoztocki) [inline notebooks/embeddings/ACP_prose_embeddings.ipynb:215]: Small question: do you know why 11 more sequences are processed than before this commit? And why it's not 1378, the total number of sequences in ACP train? The same inconsistencies occur later as well, I'm assuming it's just a logging issue on the side of ProSE?

**Comment by zunicd** [comment]
The number you see is not the exact number of sequences.There are two things running at the same time:sequence processing and writing the counter in the output.
I think the writing is slower than processing and when all sequences are processed the writing will stop.It will be always small difference. You can check the processed sequences in the h5 or pt files.I was having the same question when I ran it for the first time.
I had to rerun all notebooks due to move one level down. Without rerun many outputs will show ../ insteafd of ../../As I mentioned, the counter will stop randomly just before the end. This is why two runs will not show the same number. And this number is just when the counter has stopped, not the number of sequences.




    On Wednesday, October 5, 2022 at 10:51:21 PM EDT, Michal Roztocki ***@***.***> wrote:  
 
 


@MichalRoztocki commented on this pull request.

In notebooks/embeddings/ACP_prose_embeddings.ipynb:
>        "# embedding with pool=avg\n",
-      "# 1361 sequences processed...\r"
+      "# 1372 sequences processed...\r"

Small question: do you know why 11 more sequences are processed than before this commit? And why it's not 1378, the total number of sequences in ACP train?

File: notebooks/modelling/ACP_modelling.ipynb [hunk 2]
```diff
@@ -0,0 +1,1065 @@
...
+   "source": [
+    "The modelling loop includes the following steps:\n",
+    "\n",
+    "1. Loop through train and test embedding folders\n",
+    "2. Run the function `read_embeddings()` for train embeddings to get `X_train` and `y_train`\n",
+    "3. Run the function `read_embeddings()` for test embeddings to get `X_test` and `y_test`\n",
+    "4. Define and print the output header\n",
+    "5. Use the function `fit_tune_CV()`to to do the following:\n",
+    "   - use above defined `pipelines` and `hp_grids` dictionaries and `GridSearchCV()` to get models\n",
+    "   - save the models with `joblib`\n",
```
Comment (MichalRoztocki) [inline notebooks/modelling/ACP_modelling.ipynb:384]: I noticed you don't actually use joblib in this notebook to save the models, did you remove that line of code on purpose?

**Review by MichalRoztocki — COMMENTED** [review]
Looks good to merge for me, I just committed some typo fixes but no code changes.

File: notebooks/embeddings/ACP_prose_embeddings.ipynb (hunk 1 above)
Comment (MichalRoztocki) [reply notebooks/embeddings/ACP_prose_embeddings.ipynb:215]: Ah I see, makes sense, thanks!

**Comment by zunicd** [comment]
This is a general description that covers the activities in the called functions as well.In one of the functions models are saved with joblib.


    On Thursday, October 6, 2022 at 03:02:28 PM EDT, Michal Roztocki ***@***.***> wrote:  
 
 


@MichalRoztocki commented on this pull request.

In notebooks/modelling/ACP_modelling.ipynb:
> +   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5e22e4c4-d6b9-4abc-8305-9e330565a547",
+   "metadata": {},
+   "source": [
+    "The modelling loop includes the following steps:\n",
+    "\n",
+    "1. Loop through train and test embedding folders\n",
+    "2. Run the function `read_embeddings()` for train embeddings to get `X_train` and `y_train`\n",
+    "3. Run the function `read_embeddings()` for test embeddings to get `X_test` and `y_test`\n",
+    "4. Define and print the output header\n",
+    "5. Use the function `fit_tune_CV()`to to do the following:\n",
+    "   - use above defined `pipelines` and `hp_grids` dictionaries and `GridSearchCV()` to get models\n",
+    "   - save the models with `joblib`\n",

I noticed you don't actually use joblib in this notebook to save the models, did you remove that line of code on purpose?

**Pull request merged**